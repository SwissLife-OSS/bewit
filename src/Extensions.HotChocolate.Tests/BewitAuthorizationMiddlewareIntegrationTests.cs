using System;
using System.Threading.Tasks;
using Bewit.Extensions.HotChocolate.Validation;
using HotChocolate.Execution;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;
using Squadron;
using Xunit;

namespace Bewit.Extensions.HotChocolate.Tests
{
    public class BewitAuthorizationMiddlewareIntegrationTests
        : IClassFixture<MongoResource>
    {
        private readonly MongoResource _mongoResource;

        public BewitAuthorizationMiddlewareIntegrationTests(MongoResource mongoResource)
        {
            _mongoResource = mongoResource;
        }

        [Fact]
        public async Task Query_WhenAuthorize_Success()
        {
            // arrange
            IServiceProvider services = TestHelpers.CreateSchema<CustomPayload>();
            var payload = new CustomPayload { Email = "foo@bar.gmail.com" };
            var token = await TestHelpers.CreateToken(services, payload);

            // act
            IExecutionResult result = await TestHelpers.ExecuteQuery(services, token);

            // assert
            IBewitContext bewitContext = services.GetService<IBewitContext>();
            CustomPayload customPayload = await bewitContext.GetAsync<CustomPayload>();
            new { QueryResult = result, BewitContext = customPayload }.MatchSnapshot();
        }
    }
}
