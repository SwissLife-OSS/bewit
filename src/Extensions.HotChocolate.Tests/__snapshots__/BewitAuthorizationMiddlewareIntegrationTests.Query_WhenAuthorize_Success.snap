{
  "QueryResult": {
    "Label": null,
    "Path": null,
    "Data": {
      "foo": "bar"
    },
    "Errors": null,
    "Extensions": null,
    "ContextData": null,
    "HasNext": null
  },
  "BewitContext": {
    "Email": "foo@bar.gmail.com"
  }
}
